controllers/database_secret_handler.go

```diff
@@ -0,0 +1,210 @@
+}
+
+
+/* ------ Event Filter Functions ------ */
+
+func getDatabasesBySecret(c client.Client, secret types.NamespacedName) ([]kciv1alpha1.Database, []string, error) {
+  databaseList :=kciv1alpha1.DatabaseList{}
+  if err := c.List(context.Background(), &databaseList, &client.ListOptions{Namespace: secret.Namespace}); err != nil {
+    return nil, nil, err
+  }
+  var matchedDbs []kciv1alpha1.Database
+  var matchedDbNames []string
+  for _, database := range databaseList.Items {
+    if database.Spec.SecretName == secret.Name {
+      matchedDbs = append(matchedDbs, database)
+      matchedDbNames = append(matchedDbNames, database.Name)
+    }
+  }
+  return matchedDbs, matchedDbNames, nil
+}
+
+func isFieldUpdated(dataOld map[string][]byte, dataNew map[string][]byte, keyName string) (bool) {
+  
+  // read old and new value
+  oldValue, oldValueOk := dataOld[keyName]
+  newValue, newValueOk := dataNew[keyName]
+  
+  if !oldValueOk || !newValueOk {
+    return false // values empty or do not exist
+  }
+  
+  result := bytes.Compare(oldValue, newValue)
+  return result != 0 // true if values are not empty and updated
+}
+
+func isWatchedNamespace(watchNamespaces []string, ro runtime.Object) bool {
+  if watchNamespaces[0] == "" { // # it's necessary to set "" to watch cluster wide
+    return true // watch for all namespaces
+  }
+  // define object's namespace
+  objectNamespace := ""
+  database, isDatabase := ro.(*kciv1alpha1.Database)
+  if isDatabase {
+    objectNamespace = database.Namespace
+  } else {
+    secret, isSecret := ro.(*corev1.Secret)
+    if isSecret {
+      objectNamespace = secret.Namespace
+    } else {
+      logrus.Info("unknown object", "object", ro)
+      return false
+    }
+  }
+  
+  // check that current namespace is watched by db-operator
+  for _, ns := range watchNamespaces {
+    if ns == objectNamespace {
+      return true
+    }
+  }
+  return false
+}
+
+
+func isDatabase(ro runtime.Object) bool {
+  _, isDatabase := ro.(*kciv1alpha1.Database)
+  return isDatabase
+}
+
+
+func isObjectUpdated(e event.UpdateEvent) bool {
+  if e.ObjectOld == nil {
+    logrus.Error(nil, "Update event has no old runtime object to update", "event", e)
+    return false
+  }
+  if e.ObjectNew == nil {
+    logrus.Error(nil, "Update event has no new runtime object for update", "event", e)
+    return false
+  }
+  // if object kind is a Database check that 'metadata.generation' field ('spec' section) has been changed
+  _, isDatabase := e.ObjectNew.(*kciv1alpha1.Database)
+  if isDatabase {
+    return e.ObjectNew.GetGeneration() != e.ObjectOld.GetGeneration()
+  }
+  
+  // if object kind is a Secret check that password value has changed
+  secretNew, isSecret := e.ObjectNew.(*corev1.Secret)
+  if isSecret {
+    // if object kind is a Secret let's check that DB password inside has been updated
+    secretOld, secretOldOk := e.ObjectOld.(*corev1.Secret)
```

Comment:
You could compare secret checksum annotation here with new Secret to determin of the secret object is updated or not.

Reply:
Unfortunatelly it is not possible , because fuction [isObjectUpdated(e event.UpdateEvent)](https://github.com/vladimir22/db-operator/blob/0c2fb258609d2ae226f92aaefe3f72415c62e7f6/controllers/database_secret_handler.go#L158) works only with [event.UpdateEvent](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/event#UpdateEvent)  object related to exact single resource ( `Secret` **or** `Database`). This event object is generated only when `Secret` **or** `Database`  resource (`spec:` section) has been changed.